test/e2e/aks_autoscaler.go

```diff
@@ -62,7 +63,14 @@ func AKSAutoscaleSpec(ctx context.Context, inputGetter func() AKSAutoscaleSpecIn
 	Expect(err).NotTo(HaveOccurred())
 
 	ammp := &infrav1.AzureManagedMachinePool{}
-	err = mgmtClient.Get(ctx, client.ObjectKeyFromObject(input.MachinePool), ammp)
+	if input.isClusterClass {
+		err = bootstrapClusterProxy.GetClient().Get(ctx, types.NamespacedName{
+			Namespace: input.MachinePool.Spec.Template.Spec.InfrastructureRef.Namespace,
+			Name:      input.MachinePool.Spec.Template.Spec.InfrastructureRef.Name,
+		}, ammp)
+	} else {
+		err = mgmtClient.Get(ctx, client.ObjectKeyFromObject(input.MachinePool), ammp)
+	}
```

Comment:
Would the `isClusterClass == true` branch also work in non-cluster class scenarios?

Reply:
Actually yes I think it does, I can remove this check then